### PR TITLE
Use `time.Time` in more places

### PR DESCRIPTION
**Pull request by Tang8330**
As the title states. The goal is to eventually get rid of `ExtendedTime` altogether.

### CODE REVIEW

File: clients/shared/default_value_test.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ var dialects = []sql.Dialect{
 
 func TestColumn_DefaultValue(t *testing.T) {
 	birthday := time.Date(2022, time.September, 6, 3, 19, 24, 942000000, time.UTC)
-	birthdayExtDateTime, err := ext.ParseExtendedDateTime(birthday.Format(ext.ISO8601), ext.TimestampTZKindType)
+	birthdayExtDateTime, err := ext.ParseDateTime(birthday.Format(ext.ISO8601), ext.TimestampTZKindType)
```
Comment (danafallon) [inline clients/shared/default_value_test.go:28]: Nit: want to rename `birthdayExtDateTime` to `birthdayDateTime`?

File: lib/typing/ext/parse_test.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ func TestParseFromInterface(t *testing.T) {
 		for _, val := range vals {
 			extTime, err := ParseFromInterface(val, TimestampTZKindType)
```
Comment (danafallon) [inline lib/typing/ext/parse_test.go:19]: Rename this var too?

File: lib/typing/ext/parse_test.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ func TestParseFromInterface(t *testing.T) {
...
 			assert.NoError(t, err)
-			assert.Equal(t, val, extTime)
+			assert.Equal(t, val.(*ExtendedTime).GetTime(), extTime)
```
Comment (danafallon) [inline lib/typing/ext/parse_test.go:21]: On line 13, can we declare `vals` as `[]ExtendedTime` instead of `[]any` so we don't have to cast `val` here?

Comment (danafallon) [reply lib/typing/ext/parse_test.go:21]: Oh or will that make `ParseFromInterface` not accept it?

File: lib/typing/ext/parse_test.go [hunk 4]
```diff
@@ -40,72 +40,66 @@ func TestParseFromInterface(t *testing.T) {
...
 }
 
 func TestParseFromInterfaceTime(t *testing.T) {
 	now := time.Now()
 	for _, supportedTimeFormat := range SupportedTimeFormats {
-		et, err := ParseFromInterface(now.Format(supportedTimeFormat), TimeKindType)
+		_time, err := ParseFromInterface(now.Format(supportedTimeFormat), TimeKindType)
 		assert.NoError(t, err)
-		assert.Equal(t, TimeKindType, et.GetNestedKind().Type)
 		// Without passing an override format, this should return the same preserved dt format.
```
Comment (danafallon) [inline lib/typing/ext/parse_test.go:73]: Is this comment no longer relevant?

File: lib/typing/ext/parse_test.go [hunk 5]
```diff
@@ -40,72 +40,66 @@ func TestParseFromInterface(t *testing.T) {
...
 
 func TestParseFromInterfaceDate(t *testing.T) {
 	now := time.Now()
 	for _, supportedDateFormat := range supportedDateFormats {
-		et, err := ParseFromInterface(now.Format(supportedDateFormat), DateKindType)
+		_time, err := ParseFromInterface(now.Format(supportedDateFormat), DateKindType)
 		assert.NoError(t, err)
-		assert.Equal(t, DateKindType, et.GetNestedKind().Type)
 
 		// Without passing an override format, this should return the same preserved dt format.
```
Comment (danafallon) [inline lib/typing/ext/parse_test.go:84]: Same with this one

**Review by danafallon — APPROVED** [review]
A few nitpicks, otherwise lg!

File: lib/typing/ext/parse_test.go (hunk 3 above)
Comment (Tang8330) [reply lib/typing/ext/parse_test.go:21]: It can accept `ExtendedTime`!

File: lib/typing/ext/parse_test.go (hunk 4 above)
Comment (Tang8330) [reply lib/typing/ext/parse_test.go:73]: Yup, we don't need it anymore!

**Pull request merged**